lib/log_monitor.rb

```diff
@@ -47,15 +47,30 @@ def log_access(log_entry)
     end
   end
 
-  def check_fails_of_known_good(log_entry)
-    return if log_entry.status =~ /^[123]..$/
-
-    if @known_good_urls.include?(log_entry.path)
+  def check_fails(log_entry)
+    logstash_tags = []
+    if known_good_fail?(log_entry)
       statsd_sender.increment("known_good_fail.status_#{log_entry.status}")
-      logstash_sender.log(log_entry, "known_good_fail")
+      logstash_tags << "known_good_fail"
+    end
+
+    if cdn_fall_back?(log_entry)
+      logstash_tags << "cdn_fallback"
+    end
+
+    unless logstash_tags.empty?
+      logstash_sender.log(log_entry, logstash_tags)
     end
   end
 
+  def known_good_fail?(log_entry)
+    log_entry.status !~ /^[123]..$/ and @known_good_urls.include?(log_entry.path)
+  end
+
+  def cdn_fall_back?(log_entry)
+    !(log_entry.cdn_backend.nil? || ["", "origin"].include?(log_entry.cdn_backend))
```

Comment:
Could this line be split out to make it more readable and separate out the two checks? E.g.:

```ruby
return false if log_entry.cdn_backend.nil?

return ! ["", "origin"].include?(log_entry.cdn_backend)
```

I'm not keen on that last line, but I think that expresses the idea.